src/side-navigation/internal.tsx

```diff
@@ -412,6 +412,7 @@ function ExpandableLinkGroup({ definition, fireFollow, fireChange, activeHref, v
       variant="navigation"
       expanded={userExpanded ?? expanded}
       onChange={onExpandedChange}
+      __disableExpandChangeOnHeaderTextClick={disableExpandChangeOnHeaderTextClick}
```

Comment:
Can we do this without an internal prop? 

For example, what if we baked this behavior into the `navigation` variant of expandable section?

Reply:
I've changed from using an internal prop (for detecting the desired behaviour of the trigger to expand the section) into the `"navigation"` variant of the expandable section.

> `navigation` - Use this variant in the navigation panel with anchors and custom styled content. It doesn't have any default styles.

This is a more impactful change, but between a successful dry-run (ID: 6564992503) and our current guidelines (above), I'm confident this is the best approach.

Reply:
Let's do it!